### PR TITLE
feat: created otp input component, styled it, added the needed requir…

### PR DESCRIPTION
**Pull request by Arseniia-Damaksina**
…ements for accepting criteria

## Describe your changes

- I created an otp code component for email verification;
- Added functionality to this otp code according accepting criteria;

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/jira/software/projects/SCRUM/boards/1/backlog?selectedIssue=SCRUM-7)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
otp-code

### CODE REVIEW

File: src/components/shared/OTPInput/OTP/OTPhook.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  const handleKeyDown = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+    currentIndex: number
+  ): void => {
+    switch (event.key) {
+      case ' ':
+        event.preventDefault();
+        break;
+
+      case 'ArrowLeft':
```
Comment (lyudad) [inline src/components/shared/OTPInput/OTP/OTPhook.ts]: strings move to constants

File: src/components/shared/OTPInput/OTP/OTPhook.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        inputRefs.current[indexToEnter].value &&
+        indexToEnter < currentIndex
+      ) {
+        indexToEnter += 1;
+      } else {
+        break;
+      }
+    }
+    onChange((prev) => {
+      const otpArray = prev.split('');
```
Comment (lyudad) [inline src/components/shared/OTPInput/OTP/OTPhook.ts:101]: I assume some comments should be added, to shortly describe what is going on here

File: src/pages/VerifyEmailPage/index.tsx [hunk 3]
```diff
@@ -11,28 +12,50 @@ import RegularText from 'src/components/shared/RegularText';
 import LabelText from 'src/components/shared/LabelText';
 import TextButton from 'src/components/shared/TextButton';
 import ButtonText from 'src/components/shared/ButtonText';
+import OTPInput from 'src/components/shared/OTPInput';
 import useVerificationTimer from './VerifyEmailPageHooks';
 
 function VerifyEmailPage() {
+  const otpLengthMin = 0;
+  const otpLengthMax = 6;
```
Comment (lyudad) [inline src/pages/VerifyEmailPage/index.tsx]: move them out of the component, we don't need reinitialisation with each rerender

File: src/components/shared/OTPInput/OTP/OTPhook.ts (hunk 2 above)
Comment (Arseniia-Damaksina) [reply src/components/shared/OTPInput/OTP/OTPhook.ts:101]: Otp input is one parent component which has 6 small inputs. Parent component returns string like "123456", but inside it contains inputs values in an array like ["1", "2"...], so to make changes to the input value handler first splits the input value string to array of strings, makes changes and returns the joined string. The handlers work quite nice providing all functionality of otp: accept only one symbol, once one symbol is entered or deleted - the next input becomes focused, paste of all otp code.

**Pull request merged**